_posts/2021-02-18-survey/survey.Rmd

```diff
@@ -0,0 +1,314 @@
+
+Looking at industry sectors, we again find finance, consulting, and healthcare dominating.[^5]
+
+[^5]: In hindsight, if you were a consultant, the survey should have asked what industry sectors you were doing consulting for/in.
+
+```{r, eval = TRUE, echo = FALSE, layout="l-body-outset", fig.cap="Number of users reporting to use Spark in industry. Smaller groups not displayed."}
+knitr::include_graphics("images/spark_ind.png")
+
+```
+
+What do survey respondents do with Spark? Analyses of tabular data and time series dominate:
+
+```{r, eval = TRUE, echo = FALSE, layout="l-body-outset", fig.cap="Number of users reporting to use Spark in industry. Smaller groups not displayed."}
+knitr::include_graphics("images/spark_appl.png")
+
+```
+
+### Frameworks and skills
+
+As with deep learning, we wanted to know what language people use to do Spark. If you look at the below graphic, you see R appearing twice: once in connection with `sparklyr`, once with `SparkR`. What's that about?
+
+Both `sparklyr` and `SparkR` are R interfaces for Apache Spark, each designed and built with very different sets of priorities and, consequently, trade-offs in mind.
+
+[`sparklyr`](https://spark.rstudio.com/), one the one hand, will appeal to data scientists at home in the tidyverse, as they'll be able to use all the data manipulation interfaces they're familiar with from packages such as `dplyr`, `DBI`, `tidyr`, or `broom`.
+
+[`SparkR`](https://spark.apache.org/docs/latest/sparkr.html), on the other hand, is a light-weight R binding for Apache Spark, and bundled with the same. It's an excellent choice for practitioners who are well-versed in Apache Spark and just need a thin wrapper to access various Spark functionalities from R.
+
+```{r, eval = TRUE, echo = FALSE, layout="l-body-outset", fig.cap="Language / language bindings used to do Spark."}
+knitr::include_graphics("images/spark_lang.png")
+
+```
+
+When asked to rate their expertise in R[^6] and Spark, respectively, respondents showed similar behavior as observed for deep learning above: Most people seem to think more of their R skills than their theoretical Spark-related knowledge. However, the same caution should be exercised here as above, -- or more caution, even: The number of responses here amounted to about half of its DL counterpart.
+
+[^6]: This question only applied Spark users working in R.
+
+```{r, eval = TRUE, echo = FALSE, layout="l-body-outset", fig.cap="Self-rated skills re R and Spark."}
+knitr::include_graphics("images/spark_skills.png")
+```
+
+### Wishes and suggestions
+
+Just like with DL, Spark users were asked what could be improved, and what they were hoping for.
+
+Interestingly, answers are less "clustered" than for DL. While with DL, a few things cropped up again and again, and there were very few mentions of concrete technical features, here we see about the opposite: The great majority of wishes are concrete, technical, and often, have been uttered just a single time.
```

Comment:
```suggestion
Interestingly, answers were less "clustered" than for DL. While with DL, a few things cropped up again and again, and there were very few mentions of concrete technical features, here we see about the opposite: The great majority of wishes were concrete, technical, and often only came up once.
```

Reply:
Again, I think past tense is more appropriate here (which part of you must, too, since you slip into it by the second sentence— _cropped up…there were_).

Reply:
most definitely ;-)